### PR TITLE
EIP-2537 BLS12-381 G1 add/mul/exp and G2 add/mul support with tests

### PR DESCRIPTION
**Pull request by tersec**
https://github.com/status-im/nimbus-eth1/issues/2302

Mostly re-enabling existing code with modified gas costs

Upstream test vectors now live at https://github.com/ethereum/EIPs/tree/master/assets/eip-2537 but there's some slight reformatting already in this repo, so keeping that

These are easier to verify the test cases for. The G2 exp cases are tricker, but they're based on the same base gas cost as G2 mul, so good to have G2 mul in place for regression/consistency checks

### CODE REVIEW

File: nimbus/evm/precompiles.nim [hunk 1]
```diff
@@ -35,35 +35,29 @@ type
...
-
-proc getMaxPrecompileAddr(fork: EVMFork): PrecompileAddresses =
+func getMaxPrecompileAddr(fork: EVMFork): PrecompileAddresses =
   if fork < FkByzantium: paIdentity
   elif fork < FkIstanbul: paPairing
-  # EIP 2537: disabled
-  # reason: not included in berlin
-  # elif fork < FkBerlin: paBlake2bf
-  elif fork < FkCancun: paBlake2bf
+  elif fork < FkPrague: paBlake2bf
```
Comment (jangko) [inline nimbus/evm/precompiles.nim:54]:
```Nim
elif fork < FkCancun: paBlake2bf
elif fork < FkPrague: paPointEvaluation
```

**Comment by mratsim** [comment]
Would it be also possible to add Mgas/s similar to what I do there: https://github.com/mratsim/constantine/blob/7ea21a2e45c5e02f8aff2cfc1d94cd3191108559/benchmarks/bench_eth_evm_precompiles.nim#L63-L76

Can be tested vs Constantine with

```
git clone https://github.com/mratsim/constantine
cd constantine
CC=clang nimble bench_eth_evm_precompiles
```


The BN254 and BLS12-381 seem significantly mispriced to me given the current state of cryptographic software.

https://ethereum-magicians.org/t/eip-2537-bls12-precompile-discussion-thread/4187/76

**Comment by tersec** [comment]
> Would it be also possible to add Mgas/s similar to what I do there: https://github.com/mratsim/constantine/blob/7ea21a2e45c5e02f8aff2cfc1d94cd3191108559/benchmarks/bench_eth_evm_precompiles.nim#L63-L76
> 
> Can be tested vs Constantine with
> 
> ```
> git clone https://github.com/mratsim/constantine
> cd constantine
> CC=clang nimble bench_eth_evm_precompiles
> ```
> 
> The BN254 and BLS12-381 seem significantly mispriced to me given the current state of cryptographic software.
> 
> https://ethereum-magicians.org/t/eip-2537-bls12-precompile-discussion-thread/4187/76

It seems like various gas prices were already reduced, e.g., G1 add from 600 to 500, G2 add from 4500 to 800, and G2 mul from 55,000 to 45,000, so there's already been some consideration of this? But, alright, this seems reasonable to look at.

File: nimbus/evm/precompiles.nim (hunk 1 above)
Comment (tersec) [reply nimbus/evm/precompiles.nim]: fixed

**Pull request merged**